pyls/rpc_manager.py

```diff
@@ -0,0 +1,171 @@
+
+         Args:
+             method (str): The method name of the notification to send
+             params (dict): The payload of the notification
+         """
+        log.debug('Notify %s %s', method, params)
+        notification = JSONRPC20Request(method=method, params=params)
+        self._message_manager.write_message(notification.data)
+
+    def cancel(self, request_id):
+        """Cancel pending request handler.
+
+        Args:
+            request_id (string | number): The id of the original request
+
+        Note:
+            Request will only be cancelled if it has not begun execution.
+        """
+        log.debug('Cancel request %d', request_id)
+        try:
+            self._received_requests[request_id].cancel()
+        except KeyError:
+            log.error('Received cancel for finished/nonexistent request %d', request_id)
+
+    def consume_requests(self):
+        """ Infinite loop watching for messages from the client."""
+        for message in self._message_manager.get_messages():
+            if isinstance(message, JSONRPCBaseResponse):
+                self._handle_response(message)
+            else:
+                self._handle_request(message)
+
+    def _handle_request(self, request):
+        """Execute corresponding handler for the recieved request
+
+        Args:
+            request (JSONRPCBaseRequest): request to act upon
+
+        Note:
+            requests are handled asynchronously if the handler returns a callable, otherwise they are handle
+            synchronously by the main thread
+        """
+        if self._shutdown and request.method != 'exit':
+            return
+
+        params = request.params if request.params is not None else {}
+        try:
+            maybe_handler = self._message_handler(request.method, params)
+        except KeyError:
+            log.debug("No handler found for %s", request.method)
+            self._message_manager.write_message(
+                JSONRPC20Response(_id=request._id, error=JSONRPCMethodNotFound()._data).data)
+            return
+
+        if request._id in self._received_requests:
+            log.error('Received request %s with duplicate id', request.data)
+        elif callable(maybe_handler):
+            self._handle_async_request(request, maybe_handler)
+        elif not request.is_notification:
+            log.debug("sync request %s", request._id)
+            response = _make_response(request, result=maybe_handler)
+            self._message_manager.write_message(response.data)
+
+    def _handle_async_request(self, request, handler):
+        log.debug("async request %s", request._id)
+        future = self._executor_service.submit(handler)
+
+        if request.is_notification:
+            return
+
+        self._received_requests[request._id] = future
+
+        def did_finish_callback(completed_future):
+            if completed_future.cancelled():
+                log.debug('Cleared cancelled request %d', request._id)
```

Comment:
`s/Cleared/Clearing/`

Reply:
I think the current language is consistent with the other log lines